ext/ddtrace_profiling_native_extension/heap_recorder.c

```diff
@@ -89,9 +90,10 @@ static void heap_record_free(heap_record*);
 typedef struct {
   long obj_id;
   heap_record *heap_record;
-  live_object_data object_data;
+  // Has ownership of this, needs to clean-it-up if not transferred.
+  live_object_data *object_data;
 } object_record;
```

Comment:
Hmm, I wonder if it's time to revisit the need of `live_object_data` to exist stand-alone.

Prior to this change, it was just something we could pass a reference to, so you'd get only that part, and we didn't need to expose the whole `object_record`.

Which... is fine. But now we're doing two allocations and have to track the lifetime of two things -- is it perhaps time to inline the fields from `live_object_data` into `object_record` directly?

(And yeah, we also reference `live_object_data` from a few other structures, which makes me think we could either inline the fields or replace those structures with `object_record` as well)

Reply:
Right. I went with an intermediate approach where live_object_data still exists but as an inner data struct of `object_record`  and without a separate lifecycle. Thus I can also re-use it as part of the public API around iteration (where users should not have access to the underlying heap_record and such).

I think this is cleaner as it allows us to keep documentation centralized and easily extend it with more data later without needing to inflate arguments of `object_record_new`.